scipy/spatial/transform/_rotation.pyx

```diff
@@ -3483,13 +3483,33 @@ cdef class Rotation:
             # We first find the minimum angle rotation between the primary
             # vectors.
             cross = np.cross(b_pri[0], a_pri[0])
-            theta = atan2(_norm3(cross), np.dot(a_pri[0], b_pri[0]))
-            if theta < 1e-3:  # small angle Taylor series approximation
+            cross_norm = _norm3(cross)
+            theta = atan2(cross_norm, np.dot(a_pri[0], b_pri[0]))
+            tolerance = 1e-3  # tolerance for small angle approximation (rad)
+            R_flip = cls.identity()
+            if abs(np.pi - theta) < tolerance:
+                if cross_norm == 0:
```

Comment:
I think it's better to consider a single "near anti-parallel" case.

By doing the flip we will transform it into the "near parallel" case, which we handle robustly.

---

As for the "flip rotation" computation I suggest the following algorithm. If we want to find a rotation that flips a unit vector `x` we need to find any unit orthogonal vector to it. 

We can do it robustly by finding a component of `x` with a minimum magnitude:
```
x = [x1, x2, x3] and x3 has the minimum magnitude (for example)
y' = [-x2, x1, 0] will be orthogonal to x and have sufficiently large norm
Then y = normalize(y') is the sought orthogonal vector.
```
Then the rotation is computed from the rotation vector `pi * y`.

--- 

So the overall algorithm:

1. If `pi - theta < tol` , compute `R_flip` from `b_pri[0]`. We should use `b_pri` for that if `R_flip` is placed on the right, am I right? I am not sure about that.
2. Compute the rotation using small angle path with adjusted `cross` and `theta`.
3. Compute the final rotation as already implemented.

Reply:
The problem with choosing an arbitrary axis of rotation for the near 180 deg case is that it won't end up being the minimum angle rotation, and it messes up the overall result in my testing. Otherwise, that algorithm does work, and I while my method below for finding an orthogonal vector from unit vectors is guaranteed to work and has reasonable norms, I really like that your method doesn't involve any multiplications so I switched to it. The indexing is hard to understand, but it should be marginally more accurate.

```python
x = [x1, x2, x3]
y = [x3+1, x1, x2]
y' = np.cross(x, y)
```

Reply:
> The problem with choosing an arbitrary axis of rotation for the near 180 deg case is that it won't end up being the minimum angle rotation, and it messes up the overall result in my testing. Otherwise, that algorithm does work and I updated towards that direction.

I think I understand the issue. This 180 degree rotation looks to be the true degenerate case.

In general, why can't we use the matrix algorithm to find the rotation for two vectors? I vaguely remember, that it can give non-minimal rotation. Is that correct?

Reply:
I'm not honestly remembering the full reasoning right now, but I'm sure we covered it in our discussion when implementing this originally: https://github.com/scipy/scipy/pull/17542

Reply:
For pairs of vectors, we went this route because the Kabsch algorithm will be inexact with a finite weight (though in practice it converges quickly). This is also the code path that aligns single vectors, which needed this algorithm to find the minimal rotation rather than *a* rotation.